app/assets/javascripts/miq_tree.js

```diff
@@ -147,88 +131,168 @@ function miqTreeScrollToNode(tree, id) {
   }
 }
 
-function miqOnClickAutomate(id) {
-  miqTreeExpandNode('automate_tree', id);
-  miqJqueryRequest('/' + ManageIQ.controller + '/ae_tree_select/?id=' + encodeURIComponent(id) + '&tree=automate_tree');
-}
-
-function miqOnClickAutomateCatalog(id) {
-  miqTreeExpandNode('automate_catalog_tree', id);
-  miqJqueryRequest('/' + ManageIQ.controller + '/ae_tree_select/?id=' + encodeURIComponent(id) + '&tree=automate_catalog_tree');
-}
-
-function miqOnClickIncludeDomainPrefix() {
-  miqJqueryRequest('/' + ManageIQ.controller + '/ae_tree_select_toggle?button=domain');
-}
-
 // delete specific tree cookies
 function miqDeleteTreeCookies(tree_prefix) {
   miqTreeClearState(tree_prefix);
 }
 
 // toggle expand/collapse all nodes in tree
 function miqTreeToggleExpand(treename, expand_mode) {
-  expand_mode ? miqTreeObject(treename).expandAll() : miqTreeObject(treename).collapseAll();
+  if (isReduxTree(treename)) {
+    ManageIQ.redux.store.dispatch({namespace: treename, type: '@@tree/expandAll', value: expand_mode});
+  } else {
+    expand_mode ? miqTreeObject(treename).expandAll() : miqTreeObject(treename).collapseAll();
+  }
 }
 
-// OnClick handler for the VM Snapshot Tree
-function miqOnClickSnapshots(id) {
-  var pieces = id.split(/-/);
-  var shortId = pieces[pieces.length - 1];
-  miqJqueryRequest('/' + ManageIQ.controller + '/snap_pressed/' + encodeURIComponent(shortId), {beforeSend: true, complete: true});
+// /////////////////////////////////////////////////////////////////////////////
+// START: ONCHECK FUNCTIONS
+// /////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Function for the react tree to select checked keys.
+ * @param  {Object} tree  The tree object itself.
+ * @return {Array}        Array of keys.
+ */
+function getSelectedKeys(tree) {
```

Comment:
As this is a global, let's name it a little more descriptive:
```suggestion
function miqTreegetSelectedKeys(tree) {
```